### ISSUE TITLE
Feature Request: Add Tags to Models for Categorization and Filtering

### ISSUE THREAD

**Issue by HabebNawatha** [meta-llama/llama-stack#542]
### 🚀 Describe the new functionality needed

The GET /models/list API should include a new field, tags, in its response. The tags field will provide a way to categorize models based on their types or purposes, such as "instruct," "chat," "vision," etc.

For example, the updated response could look like this:
```json
[
  { 
    "identifier": "Llama3.2-1B-Instruct",
    "llama_model": "Llama3.2-1B-Instruct",
    "metadata": { "ollama_model": "llama3.2:1b-instruct-fp16" },
    "provider_id": "remote::ollama",
    "tags": "instruct"
  }
]
```
This enhancement allows users to filter and categorize models efficiently based on their tags.

### 💡 Why is this needed? What if we don't build it?

1. Improved Usability: Categorizing models by tags will simplify the process of identifying relevant models, especially in cases where many models are available.

2. Scalability: As the library of models grows, the tags field will allow better organization and efficient handling of diverse model types.

3. Enhanced Developer Experience: Developers can easily integrate and use the models in their workflows without needing to implement custom logic for filtering or categorizing models manually.

4. Future-Proofing: The tags field creates a foundation for future features, such as multi-tag filtering or UI improvements for model browsing.

**Comment by dineshyv** [comment]
@HabebNawatha is the expectation that these tags be populated by the user when creating the model?

**Comment by HabebNawatha** [comment]
@dineshyv yes they can populated by the user when creating the model, and in other cases it can generated using an algorithm that retrieves specific tags from already built models. For a start I think.

**Comment by hardikjshah** [comment]
Hey @HabebNawatha , we currently only have a few models and its easy to know the purpose from the names, may be we will bring this back when we have a large set and need filtering. closing the issue for now.
